src/combinatorics/simple_graph/subgraph.lean

```diff
@@ -493,7 +525,144 @@ begin
   simp only [set.mem_to_finset, mem_neighbor_set],
 end
 
-/-! ## Subgraphs of subgraphs -/
+end subgraph
+
+section mk_properties
+/-! ### Properties of `singleton_subgraph` and `subgraph_of_adj` -/
+
+variables {G : simple_graph V} {G' : simple_graph W}
+
+instance nonempty_singleton_subgraph_verts (v : V) : nonempty (G.singleton_subgraph v).verts :=
+⟨⟨v, set.mem_singleton v⟩⟩
+
+@[simp] lemma singleton_subgraph_le_iff (v : V) (H : G.subgraph) :
+  G.singleton_subgraph v ≤ H ↔ v ∈ H.verts :=
+begin
+  refine ⟨λ h, h.1 (set.mem_singleton v), _⟩,
+  intro h,
+  split,
+  { simp [h] },
+  { simp [-set.bot_eq_empty] }
+end
+
+@[simp] lemma map_singleton_subgraph (f : G →g G') {v : V} :
+  subgraph.map f (G.singleton_subgraph v) = G'.singleton_subgraph (f v) :=
+by ext; simp only [relation.map, subgraph.map_adj, singleton_subgraph_adj, pi.bot_apply,
+  exists_and_distrib_left, and_iff_left_iff_imp, is_empty.forall_iff, subgraph.map_verts,
+  singleton_subgraph_verts, set.image_singleton]
+
+@[simp] lemma neighbor_set_singleton_subgraph (v w : V) :
+  (G.singleton_subgraph v).neighbor_set w = ∅ :=
+by { ext u, refl }
+
+@[simp] lemma edge_set_singleton_subgraph (v : V) :
+  (G.singleton_subgraph v).edge_set = ∅ :=
+begin
+  ext e,
+  refine e.ind _,
+  intros,
+  simp [-set.bot_eq_empty, Prop.bot_eq_false],
```

Comment:
That will probably go better if you start with `eq_empty_of_forall_not_mem (λ e, _)`

Reply:
It's strictly longer with that lemma, but I moved most of this into some `sym2` lemmas, so that seems like a win.  The `set.bot_eq_empty` is still an issue, creating goals with terms like `¬∅ y`.